File: route/src/FooBar.ts
import { Cmd } from './Cmd';
import { RouteBase } from './RouteBase';
import { proto } from './proto/internal';

export class FooBar extends RouteBase {
public CallFoo: Cmd = Cmd.create('connector.entryHandler.callFoo', proto.Foo, proto.Bar);
public PushBar: Cmd = Cmd.create('connector.entryHandler.pushBar', null, proto.Bar);
public CallBar: Cmd = Cmd.create('connector.entryHandler.callBar', proto.Bar, proto.Foo);
}

File: unity/Assets/Plugins/Route/FooBar.cs
public class FooBar : RouteBase {
public Cmd CallFoo = new Cmd("connector.entryHandler.callFoo", typeof(Proto.Foo), typeof(Proto.Bar));
public Cmd PushBar = new Cmd("connector.entryHandler.pushBar", null, typeof(Proto.Bar));
public Cmd CallBar = new Cmd("connector.entryHandler.callBar", typeof(Proto.Bar), typeof(Proto.Foo));
}